### PR TITLE
[teams] don't show error if already in team

### LINKED ISSUE #5925: Redirect users to the members list when trying to join a team they have already joined
### Problem to solve

When a user is trying to join a team you've already joined they see the following error page.

> Redirect users to the members list when trying to join a team they have already joined

See also [relevant discussion](https://gitpod.slack.com/archives/C021Q4BL8E8/p1631009839127300) (internal). Cc @AlexTugarev 

> This was also brought up while adding this in https://github.com/gitpod-io/gitpod/pull/4490#discussion_r650154442. Cc @jankeromnes 

| Joining a team you've already joined |
|-|
| gitpod io_teams_join_inviteId=f2c69638-1d9b-483e-bad0-61a0e81204a8 |

### Proposal

Cross-posting from the [relevant discussion](https://gitpod.slack.com/archives/C021Q4BL8E8/p1631009839127300) (internal):

> Ideally, this would also include some feedback to the user using a toast component (see https://github.com/gitpod-io/gitpod/issues/3530).

> However, redirecting to the team or members list sounds like a good MVC.

### PR DESCRIPTION
**Pull request by AlexTugarev**
## Description
Directly redirect to the team if following an invitation link but already part of the team.

## Related Issue(s)

Fixes #5925

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@AlexTugarev: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **alextugarev** after the PR has been reviewed.

Associated issue: *#5925*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by gtsiolis** [comment]
Looking at this now! 👀

File: components/dashboard/src/teams/JoinTeam.tsx
Comment (gtsiolis) [inline components/dashboard/src/teams/JoinTeam.tsx:37]: **question**: This redirects you to the team's workspaces page. Do you think it's more confusing to redirect to the members list instead as described in https://github.com/gitpod-io/gitpod/issues/5925?

**Review by gtsiolis — APPROVED** [review]
UX works great, @AlexTugarev! ↩️ 

Left one comment regardingt the page we redirect below.

Approving to unblock merging but holding in case changing the redirect URL sounds better here.

/hold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 9ab15d60d0449d3d6886c3667045f192dfd2f95f</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5960#pullrequestreview-768054634" title="Approved">gtsiolis</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **alextugarev** after the PR has been reviewed.

Associated issue: *#5925*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis]
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/teams/JoinTeam.tsx
Comment (AlexTugarev) [reply components/dashboard/src/teams/JoinTeam.tsx:37]: Dunno. Let's redirect to the Members list instead. I'll ping you for a ✅

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

File: components/dashboard/src/teams/JoinTeam.tsx
Comment (AlexTugarev) [reply components/dashboard/src/teams/JoinTeam.tsx]: Done. Please have another look

Comment (gtsiolis) [reply components/dashboard/src/teams/JoinTeam.tsx]: Looking at this now! 👀

**Review by gtsiolis — APPROVED** [review]
Working like a charm, @AlexTugarev! 

**thought**: I think it makes sense to go with the redirection to the members page as :a: this is the existing behavior when joining a team for the first time and :b: this provides some feedback to the user about the link they clicked as currently there's no other visual feedback like a toast notificaton.

Approving to ublocking merging but holding in case this needs someone to take a closer look at the code changes. 

/hold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 0633ee7a5c48b1e9e055f80c18bfdabef4dada4e</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5960#pullrequestreview-768698655" title="Approved">gtsiolis</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **alextugarev** after the PR has been reviewed.

Associated issue: *#5925*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis]
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5960#issuecomment-932038958" title="Approved">AlexTugarev</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5960#pullrequestreview-768698655" title="Approved">gtsiolis</a>*

Associated issue: *#5925*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [AlexTugarev,gtsiolis]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [AlexTugarev]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
